### ISSUE TITLE
How to Provide Example Output for Multi-Shot Extraction?

### ISSUE THREAD

**Issue by nadolsw** [Serega6678/NuNER#2]
Sorry if I missed this but how does one provide multi-shot examples when using the NuNER-Extract template approach?

Should I just provide values for the keys defined within the template (I thought the model was trained to expect empty values)?

```
def predict_NuExtract(model, tokenizer, texts, template, batch_size=1, max_length=5_000, max_new_tokens=2_500):
    template = json.dumps(json.loads(template), indent=4)
    prompts = [f"""<|input|>\n### Template:\n{template}\n### Text:\n{text}\n\n<|output|>""" for text in texts]
    outputs = []
    with torch.no_grad():
        for i in range(0, len(prompts), batch_size):
            batch_prompts = prompts[i:i+batch_size]
            # Removing model.device to use CPU by default
            batch_encodings = tokenizer(batch_prompts, return_tensors="pt", truncation=True, padding=True, max_length=max_length)
            pred_ids = model.generate(**batch_encodings, max_new_tokens=max_new_tokens)
            outputs += tokenizer.batch_decode(pred_ids, skip_special_tokens=True)
    return [output.split("<|output|>")[1] for output in outputs]

# Set device to CPU
# model_name = "numind/NuExtract-v1.5"
model_name = "numind/NuExtract-tiny-v1.5"
model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=torch.bfloat16, trust_remote_code=True).eval()
tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)

text = """
...
"""

template = """{
    "Medications": [
        {
            "Drug Name": "",
            "Drug Dosage": "",
            "Drug Unit of Measure": "",
            "Drug Frequency": "",
            "Drug Route": ""
        }
    ],
    "Lab Test Results": [
        {
            "Test Date": "",
            "Test Value": "",
            "Test Unit of Measure": "",
            "Test Reference Range": ""
        }
    ]    
}"""

**Comment by Serega6678** [comment]
Hey @nadolsw 
The few-shot part is available for NuNER models only (but not NuNerZero / NuExtract which you are currently using)
